### PR TITLE
LSTNet fixes

### PR DESCRIPTION
**Pull request by kashif**
We need to use a Conv2D with appropriate kernel size

*Issue #, if available:*

*Description of changes:*
We now use the appropriate scaling and the loss takes it into account, and more importantly the Convolution has to be 2D with kernel size ` (num_series, kernel_size)`. 

We also need the hidden sizes of the two RNNs to be specified as they are not `channels`. 

I will debug the method shortly, but hopefully this describes the issues with the current implementation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/model/lstnet/_estimator.py
Comment (lostella) [inline src/gluonts/model/lstnet/_estimator.py:57]:
```suggestion
        Number of time-series (covariates)
```

**Comment by kashif** [comment]
@lostella ready for a review thanks!

**Comment by kashif** [comment]
@ehsanmok would you kindly be able to run your test on this branch? Thanks!

File: src/gluonts/model/lstnet/_network.py
Comment (ehsanmok) [inline src/gluonts/model/lstnet/_network.py:216]: This is another point of confusion at the time, since the original implementation [normalizes globally](https://github.com/laiguokun/LSTNet/blob/master/utils.py#L21), but gluonts normalization is done locally for each sliding window and it can change the results drastically and at least as an approximation is better than not normalizing I think.

File: src/gluonts/model/lstnet/_network.py
Comment (ehsanmok) [inline src/gluonts/model/lstnet/_network.py:108]: Same discussion as above, referring to the [original code](https://github.com/laiguokun/LSTNet/blob/master/models/LSTNet.py#L22).

**Review by ehsanmok — CHANGES_REQUESTED** [review]
Thanks for the fixes! Here're my comments. I tried your branch on [my code](https://github.com/awslabs/sagemaker-deep-demand-forecast) and the metrics such as RMSE improved a lot (perhaps mainly because of conv1d -> conv2d) but the visualized results *got worse* on the electricity data.

Now looking back, in general, the [original](https://github.com/laiguokun/LSTNet)pytorch implementation diverges either from the paper (an unfortunate reality of DL research!) or I needed to because of non-existent features like global normalization in gluonts.

File: src/gluonts/model/lstnet/_network.py
Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:105]: no so the number of channels of the CNN in the original code is `self.Ck` and this variable `hidC` perhaps refers to the hidden of the RNN for the convolutional output. Yes so the paper's code has `num_layers=1` but some sane numbers like `3` for it cannot hurt, but I set the `skip_rnn`'s `num_layers=1`.

Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:105]: right the RNN's input size will match the number of channels but this method takes in the hidden size of the corresponding RNN and not the size of the input.

File: src/gluonts/model/lstnet/_network.py
Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:216]: yes so lstnet divides each time series by its max in the training window and then calculates the loss on the rescaled predictions and Ys. Here its rescaled in the sliding window... so yes not sure... I say we leave it like this.

**Comment by kashif** [comment]
@ehsanmok thanks for the review. I found that for electricity dataset it helps to set the `scaling=False`. The highway network arch. of LSTNet essentially is a way for the prediction module of the network to figure out the scale, however they also normalize each time series by its max...I'll get your changes pushed asap.

File: src/gluonts/model/lstnet/_network.py
Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:108]: see my previous comment, this method takes the hidden size of the RNN not the input size which I believe Mxnet infers. The input features coming from the CNN will of course be of size `channels`.

File: src/gluonts/model/lstnet/_network.py
Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:95]: I see... yes the variable is called `self.conv1` and its `nn.Conv2d` so can get confusing... I too must admit it is a complicated architecture.

File: src/gluonts/model/lstnet/_network.py
Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:166]: I'm not sure how to test this... in my limited experiments with electricity dataset I only got a strange error when I set the `scaling=False` and I forget I think `hybridize=False`

Comment (ehsanmok) [reply src/gluonts/model/lstnet/_network.py:166]: Unfortunately, it's a [known bug](https://github.com/apache/incubator-mxnet/issues/14373) in MXNet where from some reason deferred init fails when using `scaling=False` in this case. Also by default `hybridize=True` in the gluonts trainer.

Comment (ehsanmok) [reply src/gluonts/model/lstnet/_network.py:166]: On the test side, I have done some manual hyperparameter tuning and the `min` condition came from that. So these are edge case when using different channel, skip and context length.

Comment (kashif) [reply src/gluonts/model/lstnet/_network.py:166]: ok then I can put back the `min` and run some tests.

File: test/model/lstnet/test_lstnet.py
Comment (lostella) [inline test/model/lstnet/test_lstnet.py:109]: Better to keep these thresholds loose, since otherwise tests will frequently fail (they run very short training)

```suggestion
    assert agg_metrics["ND"] < 0.5
```

**Comment by lostella** [comment]
Looks good to me, thanks @kashif 

I would just wait for the approval of @ehsanmok since he's more familiar with the implementation

**Comment by ehsanmok** [comment]
LGTM

File: src/gluonts/model/lstnet/_network.py
Comment (abdulfatir) [reply src/gluonts/model/lstnet/_network.py:166]: @kashif 3 years later I got the same strange error and finally we fixed it ;)

**Pull request merged**
